### PR TITLE
Miscellaneous additions/fixes

### PR DESCRIPTION
**Pull request by mcclure**
Merge back from my private branch of cpml

- Vector2 and 3: round to nearest int
- Bound2 and 3: Expand to encompass a point. Fix contains() unit tests. Add to init.lua
- Bound2: Fix a nil deref bug in contains()
- Vector2: Flip x and flip y functions (flip y is very useful when doing 2d coords in opengl, flip x maybe not so useful)
- Vector3: Project along a vector (this can be different from dot)
- Tests for all above

### CODE REVIEW

**Comment by shakesoda** [comment]
vector project already exists in utils, is the new one any different?

File: modules/vec2.lua [hunk 1]
```diff
@@ -321,6 +321,35 @@ function vec2.to_polar(a)
 	return radius, theta
 end
 
+-- Convert vec2 to vec3.
+-- @tparam vec2 a Vector to convert.
+-- @tparam number the new z component.
+-- @treturn vec3 Converted vector
+function vec2.append(a, z)
```
Comment (shakesoda) [inline modules/vec2.lua]: I feel like this should explicitly be something like to_vec3(a, z), append suggests happening in place

File: modules/vec2.lua [hunk 2]
```diff
@@ -321,6 +321,35 @@ function vec2.to_polar(a)
...
+-- @treturn vec2 y-flipped vector
+function vec2.flip_y(a)
+	return vec2.new(a.x, -a.y)
+end
+
+-- Round all components to nearest int.
+-- @tparam vec2 a Vector to round.
+-- @treturn vec2 Integer vector
+function vec2.round(a)
+	return vec2.new(math.floor(a.x+0.5), math.floor(a.y+0.5))
```
Comment (shakesoda) [inline modules/vec2.lua]: should this use utils.round instead? utils handles rounding negatives properly, too.

Comment (mcclure) [reply modules/vec2.lua]: yeah, i didn't know the utils module was there ^_^;

**Comment by mcclure** [comment]
A couple of these have already been fixed by other PRs… I'm going to split this up into multiple PRs instead of a mega PR soon.

**Comment by mcclure** [comment]
Some of these changes are already in master by other means, I split up any remaining changes into several independent PRs.

**Pull request closed without merging**